src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -73,36 +77,26 @@ private static String getAppendix(int number) {
         }
     }
 
-    public static String removeUnwantedCharacters(String key, boolean enforceLegalKey) {
-        if (!enforceLegalKey) {
-            // User doesn't want us to enforce legal characters. We must still look
-            // for whitespace and some characters such as commas, since these would
-            // interfere with parsing:
-            StringBuilder newKey = new StringBuilder();
-            for (int i = 0; i < key.length(); i++) {
-                char c = key.charAt(i);
-                if (KEY_UNWANTED_CHARACTERS.indexOf(c) == -1) {
-                    newKey.append(c);
-                }
-            }
-            return newKey.toString();
+    public static String removeUnwantedCharacters(String key, String unwantedCharacters) {
+        List<Character> unwantedChars = new ArrayList<>();
+        for (char ch : unwantedCharacters.toCharArray()) {
+            unwantedChars.add(ch);
         }
 
         StringBuilder newKey = new StringBuilder();
         for (int i = 0; i < key.length(); i++) {
             char c = key.charAt(i);
-            if (KEY_ILLEGAL_CHARACTERS.indexOf(c) == -1) {
+            if (!DISALLOWED_CHARACTERS.contains(c) && !unwantedChars.contains(c)) {
                 newKey.append(c);
             }
         }
```

Comment:
Please replace the code:

```java
        String newKey = key.chars()
                .filter(c -> unwantedCharacters.indexOf(c) == -1)
                .filter(c -> !DISALLOWED_CHARACTERS.contains((char) c))
                .collect(StringBuilder::new,
                        StringBuilder::appendCodePoint, StringBuilder::append)
                .toString();
```


---
I came up while thinking as follows (and stack-overflow-ing for String filter characters, ...). Used answer: https://stackoverflow.com/a/20268845/873282

Why not a simple `indexOf` thing?

```java
"abcdef".indexOf('c') != -1;
```

See https://stackoverflow.com/a/506107/873282.

Second reason: Conversion into the list is IMHO more CPU consuming than the `indexOf` call. I know, that it is not critical, but I think, the conversion (plus the readble `!unwantedChars.contains(c)`) is less readable then a simple `(unwantedChars.indexOf(c) >= 0)`

Reply:
I didn't think of it that way, thank you for your review @koppor 👍!

Reply:
You're welcome. Plese note that I used multiline comments:

![grafik](https://user-images.githubusercontent.com/1366654/81181226-6d5a4200-8fac-11ea-9a9d-0f306e01dc8c.png)

Thus, more code can be deleted. See also my new review comment.

Sorry for being picky here. I hope with these tough comments here we won't need to touch this code the next months again 🎈